### PR TITLE
Migrate ako crd to v1beta1 in akoo

### PR DESCRIPTION
**Pull request by chenlin07**
**What this PR does / why we need it**:

Migrate ako crd to v1beta1 in akoo

**Which issue(s) this PR fixes**:

Fixes #

**Describe testing done for PR**:

**Special notes for your reviewer**:

**Release note**:

```release-note

```
**New PR Checklist**

- [ ] Ensure PR contains only public links or terms
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Squash the commits in this branch before merge to preserve our git history
- [ ] If this PR is just an idea or POC, use a [Draft PR](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests) instead of a full PR
- [ ] Add appropriate [labels](https://github.com/vmware-tanzu/load-balancer-operator-for-kubernetes/labels) according to what type of issue is being addressed.

### CODE REVIEW

File: controllers/akodeploymentconfig/akodeploymentconfig_controller_avi_phase.go [hunk 1]
```diff
@@ -308,39 +308,44 @@ func (r *AKODeploymentConfigReconciler) reconcileAviInfraSetting(
...
 			},
-			Network: akov1alpha1.AviInfraSettingNetwork{
+			Network: akov1beta1.AviInfraSettingNetwork{
 				VipNetworks: vipNetwork,
 			},
-			L7Settings: akov1alpha1.AviInfraL7Settings{
+			L7Settings: akov1beta1.AviInfraL7Settings{
 				ShardSize: shardSize,
 			},
+			NSXSettings: akov1beta1.AviInfraNSXSettings{
```
Comment (XudongLiuHarold) [inline controllers/akodeploymentconfig/akodeploymentconfig_controller_avi_phase.go:346]: Can we just remove this `NSXSettings` struct?

**Review by XudongLiuHarold — COMMENTED** [review]
Please also make sure run
```
go mod tidy
```

File: controllers/akodeploymentconfig/akodeploymentconfig_controller_avi_phase.go (hunk 1 above)
Comment (chenlin07) [reply controllers/akodeploymentconfig/akodeploymentconfig_controller_avi_phase.go]: Try to create an adc in testbed, stills got error
```
2023-10-23T09:49:03.003Z	ERROR	controllers.AKODeploymentConfig	failed to reconcile AKODeploymentConfig	{"AKODeploymentConfig": {"name":"install-ako-for-test-aviinfrasetting"}, "error": "AviInfraSetting.ako.vmware.com \"install-ako-for-test-aviinfrasetting-ais\" is invalid: spec.nsxSettings.t1lr: Required value", "errorCauses": [{"error": "AviInfraSetting.ako.vmware.com \"install-ako-for-test-aviinfrasetting-ais\" is invalid: spec.nsxSettings.t1lr: Required value", "errorCauses": [{"error": "AviInfraSetting.ako.vmware.com \"install-ako-for-test-aviinfrasetting-ais\" is invalid: spec.nsxSettings.t1lr: Required value"}]}]}
```

File: controllers/akodeploymentconfig/akodeploymentconfig_controller_avi_phase.go [hunk 2]
```diff
@@ -290,59 +293,51 @@ func (r *AKODeploymentConfigReconciler) reconcileAviInfraSetting(
...
-		if apierrors.IsNotFound(err) {
-			log.Info("AVIInfraSetting doesn't exist, start creating it")
-			return res, r.Create(ctx, newAviInfraSetting)
-		}
-		log.Error(err, "Failed to get AVIInfraSetting, requeue")
-		return res, err
-	}
-	newAviInfraSetting.Spec.DeepCopyInto(&aviInfraSetting.Spec)
-	return res, r.Update(ctx, aviInfraSetting)
+	return res, r.Patch(ctx, r.createAviInfraSetting(adc), client.Apply, client.ForceOwnership, client.FieldOwner("akodeploymentconfig-controller"))
```
Comment (chenlin07) [inline controllers/akodeploymentconfig/akodeploymentconfig_controller_avi_phase.go:296]: The value in fieldowner here should be name of controller

**Review by XudongLiuHarold — DISMISSED** [review]
LGTM， thanks

File: controllers/akodeploymentconfig/akodeploymentconfig_controller_avi_phase.go [hunk 3]
```diff
@@ -290,59 +293,51 @@ func (r *AKODeploymentConfigReconciler) reconcileAviInfraSetting(
...
 				VipNetworks: vipNetwork,
 			},
-			L7Settings: akov1alpha1.AviInfraL7Settings{
+			L7Settings: akov1beta1.AviInfraL7Settings{
 				ShardSize: shardSize,
 			},
 		},
 	}
+	u := &unstructured.Unstructured{}
+	u.Object, _ = runtime.DefaultUnstructuredConverter.ToUnstructured(aviinfraSetting)
```
Comment (lubronzhan) [inline controllers/akodeploymentconfig/akodeploymentconfig_controller_avi_phase.go:339]: Better to throw error here

**Review by XudongLiuHarold — APPROVED** [review]
LGTM

**Pull request merged**